api/python/provisioner/commands/sw_rollback.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
+from typing import Type
+
+from ._basic import RunArgs, CommandParserFillerMixin
+from ..vendor import attr
+from ..config import ALL_MINIONS
+from ..errors import (
+    BadPillarDataError,
+    SWRollbackError,
+)
+from ..pillar import (
+    PillarKey,
+    PillarResolver
+)
+from ..salt import (
+    StatesApplier,
+    YumRollbackManager,
+    local_minion_id
+)
+from ..commands import _apply_provisioner_config
+from ..salt_master import config_salt_master
+from ..salt_minion import config_salt_minions
+from .. import inputs, values
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class RunArgsSWRollback:
+    target_version: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "CORTX version to rollback "
+                "in case yum rollback is not performed"
+            }
+        },
+        default=None
+    )
+    targets: str = RunArgs.targets
+
+
+@attr.s(auto_attribs=True)
+class SWRollback(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsSWRollback
+
+    @staticmethod
+    def _rollback_component(component, targets):
+        state_name = "components.{}.rollback".format(component)
+        try:
+            logger.info(
+                "Restoring {} configuration on {}".format(component, targets)
+            )
+            StatesApplier.apply([state_name], targets)
+        except Exception:
+            logger.exception(
+                "Failed to restore {} on {}".format(component, targets)
+            )
+            raise
+
+    def run(self, target_version=None, targets=ALL_MINIONS):
+
+        local_minion = local_minion_id()
+
+        upgrade_path = PillarKey('upgrade')
+        upgrade_pillar = PillarResolver(local_minion).get([upgrade_path])
+        upgrade_pillar = upgrade_pillar[local_minion][upgrade_path]
+
+        try:
+            if target_version:
+                yum_txn_ids = upgrade_pillar.get('yum_snapshots', {})
+
```

Comment:
updated yum_txn_ids with yum_snapshots since we haven't filtered pillar for specific cortx_version
added this check here
 https://github.com/Seagate/cortx-prvsnr/pull/901/files#diff-f23ccf652739101b0f4c033f04db81a744388ed6f683456759d71c4f0e759fa9R109